### DISCUSSION TITLE
How not to benchmark!

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#112] [General]
# How not to benchmark!

I came across a recent LinkedIn post about the let statement in LINQ and it's performance implication. And in typically influencer fashion it out right claimed that using let in LINQ is a bad idea and should be avoided. But is it a bad idea?

https://steven-giesel.com/blogPost/98458f74-5205-4b2b-9f5b-535e34ec2fea

**Comment by runehalfdan** [comment]
The queries aren't equal. When using 'let', both where-clauses are evaluated. When not using 'let', the where-clause can be short-circuited

**Reply by linkdotnet** [reply]
That's true and will lead to an increased execution time. Maybe I will spin up the test to do `where isLastNameHaris & isFirstNameAdmir` (bitweise and) which is more similar to `let` version.

**Comment by LwsSt** [comment]
The test here is using an in-memory List<T>, I wonder if there would be any impact if the data source was an IQueryable<T> instead?

**Reply by linkdotnet** [reply]
Interesting question. Maybe I can spin up the same test with an (in-memory) SQLite Database.
